checks.d/mesos_master.py

```diff
@@ -0,0 +1,246 @@
+        })
+
+    def _status_code_event(self, url, r, aggregation_key):
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'Invalid reponse code for %s' % url,
+            'msg_text': '%s returned a status of %s' % (url, r.status_code),
+            'aggregation_key': aggregation_key
+        })
+
+    def _get_json(self, url, timeout):
+        # Use a hash of the URL as an aggregation key
+        aggregation_key = md5(url).hexdigest()
+        tags = ["url:%s" % url]
+        msg = None
+        status = None
+        try:
+            r = requests.get(url, timeout=timeout)
+            if r.status_code != 200:
+                self._status_code_event(url, r, aggregation_key)
+                status = AgentCheck.CRITICAL
+                msg = "Got %s when hitting %s" % (r.status_code, url)
+            else:
+                status = AgentCheck.OK
+                msg = "Mesos master instance detected at %s " % url
+        except requests.exceptions.Timeout as e:
+            # If there's a timeout
+            self._timeout_event(url, timeout, aggregation_key)
+            msg = "%s seconds timeout when hitting %s" % (timeout, url)
+            status = AgentCheck.CRITICAL
+        except Exception as e:
+            msg = e.message
+            status = AgentCheck.CRITICAL
+        finally:
+            if self.SERVICE_CHECK_NEEDED:
+                self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags,
+                                   message=msg)
+                self.SERVICE_CHECK_NEEDED = False
+            if status is AgentCheck.CRITICAL:
+                self.warning(msg)
+                return None
+
+        return r.json()
+
+    def _get_master_state(self, url, timeout):
+        return self._get_json(url + '/state.json', timeout)
+
+    def _get_master_stats(self, url, timeout):
+        if self.version >= Version('0.22.0'):
+            endpoint = '/metrics/snapshot'
+        else:
+            endpoint = '/stats.json'
+        return self._get_json(url + endpoint, timeout)
+
+    def _get_master_roles(self, url, timeout):
+        return self._get_json(url + '/roles.json', timeout)
+
+    def _check_leadership(self, url, timeout):
+        json = self._get_master_state(url, timeout)
+
+        if json is not None:
+            self.version = Version(json['version'])
+            if json['leader'] == json['pid']:
+                self.leader = True
+        else:
+            self.leader = False
+        return json
+
+    def check(self, instance):
+        if 'url' not in instance:
+            raise Exception('Mesos instance missing "url" value.')
+
+        url = instance['url']
+        instance_tags = instance.get('tags', [])
+        default_timeout = self.init_config.get('default_timeout', 5)
+        timeout = float(instance.get('timeout', default_timeout))
+
+        json = self._check_leadership(url, timeout)
```

Comment:
You shouldn't use json as a variable name as it's also a python module name. Also it's not a very explicit variable name.